### ISSUE TITLE
Bug - Select - validated

### ISSUE THREAD

**Issue by dlabrecq** [patternfly/patternfly-react#10025]
The v4 `Select` supported a `validated` property; however, it has been omitted for v5. The [design guidelines](https://www.patternfly.org/components/menus/select/design-guidelines#select-lists-with-validation) currently state the "Select component offers three validation-state presentations", but I don't see how the ["validated selections" example](https://www.patternfly.org/components/menus/select/react-deprecated#validated-selections) is achieved in v5?

We've been using validation in Cost Management, but are blocked from updating to the v5 `Select`.

**Cost Management snapshot**
<img width="391" alt="Screenshot 2024-01-19 at 9 19 55 AM" src="https://github.com/patternfly/patternfly-react/assets/17481322/d8291539-549f-4c2c-8adc-1f033a589fd0">

**Comment by dlabrecq** [comment]
Related  https://github.com/patternfly/patternfly/issues/6242

**Comment by rbertucat** [comment]
Was also opened here at some point: https://github.com/patternfly/patternfly/issues/5917

**Comment by mcoker** [comment]
Core work is done here https://github.com/patternfly/patternfly/pull/6434

TL;DR
* The parent component element (`.pf-v5-c-menu-toggle`) gets a status class (`.pf-m-success/warning/danger`)
* Adds a new `.pf-v5-c-menu-toggle__icon` element that holds the status icon
  * This is rendered in `.pf-v5-c-menu-toggle__controls`, next to the caret icon
* We may want to add something so screen readers announce the status - https://github.com/patternfly/patternfly/pull/6434#pullrequestreview-1946422952
